### PR TITLE
YETUS-972. Make javadoc phase and goal configurable in maven.sh

### PR DESCRIPTION
**Pull request by aajisaka**
JIRA: https://issues.apache.org/jira/browse/YETUS-972

### CODE REVIEW

**Comment by busbey** [comment]
Do you want to be able to use multiple goals? Like the example from HDFS-15424 was process-sources followed by a no fork javadoc?

To do things like that the cli arg needs to take a list of goals.

File: precommit/src/main/shell/test-patch.d/maven.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ MAVEN_CUSTOM_REPOS=false
 MAVEN_CUSTOM_REPOS_DIR="@@@WORKSPACE@@@/yetus-m2"
 MAVEN_DEPENDENCY_ORDER=true
 MAVEN_FOUND_ROOT_POM=false
-MAVEN_JAVADOC="javadoc:javadoc"
+MAVEN_JAVADOC_GOALS="javadoc:javadoc"
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/maven.sh:29]: This should be an array

**Comment by aajisaka** [comment]
> Do you want to be able to use multiple goals? 

Yes. Updated the patch to allow multiple goals from CLI. Thank you @busbey.

File: precommit/src/main/shell/test-patch.d/maven.sh [hunk 2]
```diff
@@ -109,6 +111,12 @@ function maven_parse_args
         delete_parameter "${i}"
         MAVEN_DEPENDENCY_ORDER=${i#*=}
       ;;
+      --mvn-javadoc-goals=*)
+        delete_parameter "${i}"
+        MAVEN_JAVADOC_GOALS=${i#*=}
+        MAVEN_JAVADOC_GOALS=${MAVEN_JAVADOC_GOALS//,/ }
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/maven.sh:117]: This should use `yetus_comma_to_array`

File: precommit/src/main/shell/test-patch.d/maven.sh [hunk 3]
```diff
@@ -335,7 +343,7 @@ function maven_modules_worker
       modules_workers "${repostatus}" distclean clean -DskipTests=true
     ;;
     javadoc)
-      modules_workers "${repostatus}" javadoc clean javadoc:javadoc -DskipTests=true
+      modules_workers "${repostatus}" javadoc clean "${MAVEN_JAVADOC_GOALS}" -DskipTests=true
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/maven.sh:346]: This should be `"${MAVEN_JAVADOC_GOALS[@]}"`

**Comment by aajisaka** [comment]
Thank you for your quick review! Updated.

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-112/1/console in case of problems.

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 24s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  shellcheck  |   0m  3s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  shelldocs  |   0m  0s |  There were no new shelldocs issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |   1m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-112/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/112 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 97edb8290b13 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 742d816 |
| Max. process+thread count | 41 (vs. ulimit of 2000) |
| modules | C: precommit U: precommit |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-112/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by aajisaka** [comment]
Merged. Thank you @busbey !

**Pull request merged**